SUPPORTING_NEW_SERVICES.md

```diff
@@ -0,0 +1,108 @@
+
+* Batching: The [`BatchResult`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/dns-alpha-batch/gcloud-java-core/src/main/java/com/google/cloud/BatchResult.java) class provides a simple way for users to combine RPC calls for performance enhancement.  API for services that support batching should implement a batch class that contains methods similar to the methods in the `Service.java` subclass, but instead return a subclass of `BatchResult`.  Also provide an SPI-layer class to collect batch requests and submit the batch.
+
+Notes/reminders:
+* API layer classes should be located in the package `com.google.cloud.servicename`, where "servicename" corresponds to the name of the Cloud service.
+* Override the `ServiceOptions.defaultRetryParams()` method in your service's options class to align with the Service Level Agreement (SLA) given by the underlying service.  See #857 and #860 for context.
+* See conventions about overriding the `equals` and `hashCode` methods in the discussion of #892.
+* While not all fields for model objects need to be exposed to the user, `gcloud-java` services should get and set all relevant fields when making RPC calls to the Cloud service's API.  For example, since the `parent` field of Cloud Resource Manager Project objects is in alpha (at the time this is written) and not available to most users, `gcloud-java-resourcemanager` gets and sets the parent when interacting with the Cloud Resource Manager, but does not expose the parent to users.  As a result, the user won't accidentally unset the parent when updating a project.
+* Be aware of differences in "update" behavior and name update/replace methods accordingly in your API.  See #321 for context.
+* Do not expose third party libraries in the API.  This has been a design choice from the beginning of the project, and all existing `gcloud-java` services adhere to this convention.
+* Member variable getters and builder setters should not use the prefixes "get" or "set".
+* Any service-generated IDs for model objects should be named `generatedId()`.
+
+#### SPI layer
+
+The SPI layer classes should be located in the package `com.google.cloud.servicename.spi`.  In most cases, the SPI layer should contain at least three classes:
+* An RPC factory interface (allows for the implementation to be loaded via the `java.util.ServiceLoader`).
+* An RPC interface that contains all RPC methods.
+* A default RPC implementation.
```

Comment:
Should we have a reminder to include the new spi in [here](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/pom.xml#L389)?